.jenkins/doc_build

```diff
@@ -18,7 +18,7 @@ pipeline {
       steps {
         sh '''
           source "${CONDA_HOME}/etc/profile.d/conda.sh"
-          conda create -y -p "${CONDA_ENV}" python=3.9
+          conda create -y -p "${CONDA_ENV}" python=3.9 poetry>=1.3
```

Comment:
I believe we installed poetry through pipx on all the CI machines. I think we need to manually upgrade on all of them for this PR to work, right ?

Reply:
I thought the Conda version would take priority over the one installed through `pipx`, but maybe not 🤔

Reply:
I have reverted the modifications on the Jenkins CI files.